core/tools.js

```diff
@@ -1950,6 +1950,40 @@
 				return ret;
 			},
 
+			/**
+			 * Returns the first element in the array for which the given callback `fn` returns true.
+			 * If there is no such element `undefined` is returned.
```

Comment:
Actually I'd see this part in `@returns` part as otherwise it is empty, which is not optimal.

So it would be something like:
```
* @returns {*} First matched value, `undefined` otherwise.
```